### ISSUE TITLE
Frontend implementation for type specific grouping options

### ISSUE THREAD

**Issue by pavish** [centerofci/mathesar#1073]
## Problem
Users should be able to view and apply type specific grouping for each column.

## Proposed solution
We have a set of backend tickets to implement grouping options for each type. A few:
* https://github.com/centerofci/mathesar/issues/408
* https://github.com/centerofci/mathesar/issues/411
* https://github.com/centerofci/mathesar/issues/384

The backend would provide the set of possible grouping options as mentioned in the above tickets.

The frontend should make use of those options and display them accordingly, and allow the user to group records using them.

## Additional context
Blocked by:
- #408 
- #419 
- #414 
- #432 
- #384 
- #411 
- #429

**Comment by pavish** [comment]
@kgodey Do we need to implement additional grouping options for the live demo or the public release?

Currently, we have distinct grouping for all types, and types such as email, uri, date/time etc., have additional preproc functions.

I think ranged grouping and more preproc functions for types like duration can be safely deferred, and we could move this issue to the `2023 or later` milestone.

**Comment by kgodey** [comment]
@pavish Can you make a list of features that you're proposing deferring? I'm not sure what this means:

> preproc functions for types like duration

Even better than that would be to close this issue and make separate issues for each grouping type that was on our plan but hasn't been implemented yet. We can then prioritize each separately.

**Comment by pavish** [comment]
I'm marking this issue as closed since we have specific grouping per type implemented already.

Types that do not have custom grouping options are tracked separately. The tickets are mentioned in the description.

The frontend currently stubs grouping options instead of receiving it from the backend, but that is out-of-scope of this issue and is related to https://github.com/centerofci/mathesar/issues/1983
